simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/StandardExports.java

```diff
@@ -100,69 +98,33 @@ public StandardExports() {
   }
 
   static Long callLongGetter(String getterName, Object obj) throws NoSuchMethodException {
-    Method method = obj.getClass().getMethod(getterName);
-    return (Long) callMethod(method, obj).getValue();
+    return callLongGetter(obj.getClass().getMethod(getterName), obj);
   }
 
   /**
    * Attempts to call a method either directly or via one of the implemented interfaces.
    */
-  static ReturnValue callMethod(Method method, Object obj) {
+  static Long callLongGetter(Method method, Object obj) {
     try {
       method.setAccessible(true);
-      return new ReturnValue(method.invoke(obj));
+      return (Long) method.invoke(obj);
     } catch (Exception e) {
-      LOGGER.log(Level.FINE,
-          "Invocation failed on " + method.getDeclaringClass().getName() + "." + method.getName(),
-          e);
+      // Expected, setAccessible() might fail across Java 9 module boundaries.
     }
 
     for (Class<?> clazz : method.getDeclaringClass().getInterfaces()) {
-      Method interfaceMethod;
       try {
-        interfaceMethod = clazz.getMethod(method.getName(), method.getParameterTypes());
+        Method interfaceMethod = clazz.getMethod(method.getName(), method.getParameterTypes());
+        Long result = callLongGetter(interfaceMethod, obj);
```

Comment:
Apparently there is no constraint on implementing 2 interfaces defining the exact same method. The constraint is on 2 methods with the same parameters but different return types (obvious why that wouldn't work).

Still, the tree is finite and visiting more than a simple path is an edge case.